src/scripts/plugin/videojs.vast.vpaid.js

```diff
@@ -495,6 +499,15 @@ module.exports = function VASTPlugin(options) {
 
   function trackAdError(error, vastResponse) {
     player.trigger({type: 'vast.adError', error: error});
+    if (error && error.code === 402) {
+      // we care only ad start timeout error, all other errors will track in VAST integrator
+      if (player.vast && player.vast.trackError) {
+        player.vast.trackError(error, player.vast.vastResponse);
+      }
+      else {
+        logger.warn ('WARNING: Cannot track ad start timeout error because VAST XML is not parsed yet');
+      }
+    }
     cancelAds();
     logger.error ('AD ERROR:', error.message, error, vastResponse);
```

Comment:
I would move this line to before the new block fires - up to line 502 - so that an engineer debugging will see the actual error in the console right above the warning that the tracking pixel didn't fire. Otherwise all the traces from the cancelAds() call will show up between the error and the warning, making it difficult to associate them to each other.

Reply:
Did it.